src/app/app.state.ts

```diff
@@ -6,16 +7,20 @@ import { productReducer, ProductState } from './products/product.reducers';
 export interface User {
   login_token: string | null;
   refresh_token: string | null;
+  isLoading: boolean;
```

Comment:
When considering null versus undefined as the default/initial state of an NgRx store state, undefined is generally the better choice. Here's why:

Semantics:

undefined indicates that a value hasn't been assigned yet
null explicitly represents the absence of any object value


TypeScript handling:

TypeScript treats undefined as the default value for optional properties
This aligns well with NgRx's pattern of initializing state


Performance:

undefined is slightly more performant as it doesn't require explicit assignment


Consistency:

Using undefined maintains consistency with JavaScript's default behavior for uninitialized variables


Cleaner checks:

Checking for undefined is often more straightforward, especially with optional chaining (?.)